src/dune_engine/action.ml

```diff
@@ -21,7 +21,7 @@ module Make
 struct
   include Ast
 
-  let run prog args = Run (prog, args)
+  let run prog args = Run (prog, Array.Immutable.of_list args)
```

Comment:
Should this change to take an array as the argument, to encourage the use of arrays in the rules?

Reply:
(Maybe this should be done in a separate PR.)

Reply:
I tried this, but it's a little problematic because:

* It's not clear if there's any benefits as there's so many layers of conversion for all these flags already
* Quite a bit of code will have to change.

Will leave a TODO for this.